src/array_value.cpp

```diff
@@ -228,31 +228,69 @@ ValuePtr ArrayValue::map(Env *env, Block *block) {
     return new_array;
 }
 
-// TODO: accept integer and return array
-ValuePtr ArrayValue::first(Env *env) {
-    if (size() > 0) {
+ValuePtr ArrayValue::first(Env *env, ValuePtr n) {
+    auto has_count = n != nullptr;
+
+    if (!has_count) {
+        if (is_empty())
+            return NilValue::the();
+
         return (*this)[0];
```

Comment:
I copied the usage of `(*this)[0]` which only seems to add an assert but since we have checked it is non empty is that necessary? 
Same for below we (should) know the index is correct and can thus go directly to `m_vector`.

Reply:
You're correct, it's unneeded, and I've probably been inconsistent in using `[]` vs going directly to `m_vector`. Your call.